fs-storage/src/file_storage.rs

```diff
@@ -0,0 +1,223 @@
+use std::fmt::Debug;
+use std::fs::{self, File};
+use std::io::{BufRead, BufReader, BufWriter, Write};
+use std::str::FromStr;
+use std::time::SystemTime;
+use std::{
+    collections::BTreeMap,
+    fmt::Display,
+    path::{Path, PathBuf},
+};
+
+use data_error::{ArklibError, Result};
+
+const STORAGE_VERSION: i32 = 2;
+const STORAGE_VERSION_PREFIX: &str = "version ";
+const KEY_VALUE_SEPARATOR: char = ':';
+
+pub struct FileStorage {
+    label: String,
+    path: PathBuf,
+    timestamp: SystemTime,
+}
```

Comment:
I am curious what `label` represents here.
I guess also `timestamp` is the latest modified time for the whole `FileStorage`  instance?

Reply:
@twitu

Reply:
`label` is more of telemetry/debugging/logging thing ported from the Kotlin implementations. The logs for file storage operations have the label. It probably helps understanding the logs if an application uses multiple file storage instances.

Yes `timestamp` is the latest modified time for the underlying file. It is updated when the file is written to or when the file is read from.

Reply:
> The logs for file storage operations have the label. It probably helps understanding the logs if an application uses multiple file storage instances.

makes perfect sense